### PR TITLE
Bug Fix - Filenames are incorrect for Tribal nations

### PR DESCRIPTION
**Pull request by andrewedstrom**
[#179786305]

### CODE REVIEW

File: src/main/java/org/codeforamerica/shiba/output/caf/FilenameGenerator.java [hunk 1]
```diff
@@ -31,14 +33,27 @@
...
+    this.tribalNations = tribalNations;
   }
 
   public String generatePdfFilename(Application application, Document document) {
-    String prefix = getSharedApplicationPrefix(application, document);
+    String dhsProviderId = countyMap.get(application.getCounty()).getDhsProviderId();
+    String prefix = getSharedApplicationPrefix(application, document, dhsProviderId);
+    String programs = getProgramCodes(application);
+    String pdfType = document.toString();
+    return "%s%s_%s.pdf".formatted(prefix, programs, pdfType);
```
Comment (sree-cfa) [inline src/main/java/org/codeforamerica/shiba/output/caf/FilenameGenerator.java:50]: Could call the new method? 

```
RoutingDestination rd = countyMap.get(application.getCounty());
return generatePdfFilenameForRoutingDestination(application, document, rd)
```

**Comment by sree-cfa** [comment]
@andrewedstrom Do we need to do this for uploaded documents too?

File: src/main/java/org/codeforamerica/shiba/output/caf/FilenameGenerator.java (hunk 1 above)
Comment (andrewedstrom) [reply src/main/java/org/codeforamerica/shiba/output/caf/FilenameGenerator.java:50]: ooh good call

**Comment by andrewedstrom** [comment]
@sree-cfa oof good point, that's a pretty serious oversight

**Comment by sree-cfa** [comment]
LGTM! Appreciate the comments 👍

**Review by sree-cfa — APPROVED** [review]

**Pull request merged**